### PR TITLE
enable to dis/connect terminals and branches

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the new behavior (if this is a feature change)?**
enable to dis/connect terminals and branches with update methods in network

### CODE REVIEW

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java [hunk 1]
```diff
@@ -869,49 +869,105 @@ static String getPhaseTapChangerRegulatedSide(TwoWindingsTransformer transformer
...
 
     static NetworkDataframeMapper branches() {
-        return NetworkDataframeMapperBuilder.ofStream(Network::getBranchStream, NetworkDataframes::getT2OrThrow)
+        return NetworkDataframeMapperBuilder.ofStream(Network::getBranchStream, getOrThrow(Network::getBranch, "Branch"))
                 .stringsIndex("id", Branch::getId)
                 .strings("type", branch -> branch.getType().toString())
                 .strings("voltage_level1_id", branch -> branch.getTerminal1().getVoltageLevel().getId())
                 .strings("bus1_id", branch -> branch.getTerminal1().getBusView().getBus() == null ? "" :
                         branch.getTerminal1().getBusView().getBus().getId())
+                .booleans("connected_1", branch -> branch.getTerminal1().isConnected(),
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java]: we should remove the underscore for consistency (`connected1`), same for side 2

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java [hunk 2]
```diff
@@ -869,49 +869,105 @@ static String getPhaseTapChangerRegulatedSide(TwoWindingsTransformer transformer
...
+        if (connectable == null) {
+            throw new PowsyblException("connectable " + id + " not found");
+        }
+        String side = dataframe.getStringValue("element_side", index).orElse(null);
+        if (side == null) {
+            if (connectable instanceof Branch || connectable instanceof ThreeWindingsTransformer) {
+                throw new PowsyblException("side must be provided for this element");
+            }
+            return connectable.getTerminals().get(0);
+        } else if (side.equals("ONE")) {
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java]: It would be better with an enum than with strings

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java [hunk 3]
```diff
@@ -869,49 +869,105 @@ static String getPhaseTapChangerRegulatedSide(TwoWindingsTransformer transformer
...
+        String id = dataframe.getStringValue("element_id", index)
+                .orElseThrow(() -> new IllegalArgumentException("element_id column is missing"));
+        Connectable<?> connectable = network.getConnectable(id);
+        if (connectable == null) {
+            throw new PowsyblException("connectable " + id + " not found");
+        }
+        String side = dataframe.getStringValue("element_side", index).orElse(null);
+        if (side == null) {
+            if (connectable instanceof Branch || connectable instanceof ThreeWindingsTransformer) {
+                throw new PowsyblException("side must be provided for this element");
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java]: We could add the id in the message so that the user knows which is the element

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java [hunk 4]
```diff
@@ -869,49 +869,105 @@ static String getPhaseTapChangerRegulatedSide(TwoWindingsTransformer transformer
...
-                        ((Injection) connectable).getTerminal().getBusView().getBus().getId())
+                .strings("voltage_level_id", connectable -> ((Injection<?>) connectable).getTerminal().getVoltageLevel().getId())
+                .strings("bus_id", connectable -> ((Injection<?>) connectable).getTerminal().getBusView().getBus() == null ? "" :
+                        ((Injection<?>) connectable).getTerminal().getBusView().getBus().getId())
                 .build();
     }
 
     static NetworkDataframeMapper branches() {
-        return NetworkDataframeMapperBuilder.ofStream(Network::getBranchStream, NetworkDataframes::getT2OrThrow)
+        return NetworkDataframeMapperBuilder.ofStream(Network::getBranchStream, getOrThrow(Network::getBranch, "Branch"))
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java:893]: Not part of this PR but it would be better with an enum here too

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java (hunk 2 above)
Comment (EtienneLt) [reply java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java]: I'm not sure because there are enums for branch and enums for three windings transformers and if we add an enum here we will have more code

File: java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java (hunk 4 above)
Comment (EtienneLt) [reply java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java:893]: why do we need an enum here for ?

Comment (sylvlecl) [reply java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java:893]: Mainly so that people don't have to go and read the method implementation to know what are the possible values

**Review by sylvlecl — APPROVED** [review]
